File: given_parser/ast/NewArrayExpression.java
package ast;

import java.util.Hashtable;
import java.util.LinkedHashMap;
import java.util.List;

public class NewArrayExpression
    extends AbstractExpression
{
    private final String size;

    public NewArrayExpression(int lineNum, String size)
    {
        super(lineNum);
        this.size = size;
    }
}

File: given_parser/myasts/NewArrayExpression.cpp
#include "NewArrayExpression.hpp"

namespace ast {

// Constructor
NewArrayExpression::NewArrayExpression(int lineNum, const std::string& size)
    : AbstractExpression(lineNum), size(size) {}

// Getter method for size
const std::string& NewArrayExpression::getSize() const {
    return size;
}

} // namespace ast

